src/DotNetWorker/Hosting/WorkerMiddlewareWorkerApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Reflection;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Middleware;
+using Microsoft.Azure.Functions.Worker.OutputBindings;
+using Microsoft.Azure.Functions.Worker.Pipeline;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Extensions.Hosting
+{
+    /// <summary>
+    /// Provides extension methods to work with Worker Middleware against a <see cref="IHostBuilder"/>.
+    /// </summary>
+    public static class WorkerMiddlewareWorkerApplicationBuilderExtensions
+    {
+        /// <summary>
+        /// Configures the <see cref="IFunctionsWorkerApplicationBuilder"/> to use the default set of middleware used by the worker, in the following order:
+        /// <list type="number">
+        ///     <item>
+        ///         <description><see cref="OutputBindingsMiddleware"/></description>
+        ///     </item>
+        ///     <item>
+        ///         <description><see cref="FunctionExecutionMiddleware"/></description>
+        ///     </item>
+        /// </list>
+        /// </summary>
+        /// <param name="builder">The <see cref="IFunctionsWorkerApplicationBuilder"/> to configure.</param>
+        /// <returns>The configured <see cref="IFunctionsWorkerApplicationBuilder"/>.</returns>
+        public static IFunctionsWorkerApplicationBuilder UseDefaultWorkerMiddleware(this IFunctionsWorkerApplicationBuilder builder)
+        {
+            return builder.UseOutputBindingsMiddleware()
+                .UseFunctionExecutionMiddleware();
+        }
+
+        /// <summary>
+        /// Configures the <see cref="IFunctionsWorkerApplicationBuilder"/> to use the default <see cref="FunctionExecutionMiddleware"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="IFunctionsWorkerApplicationBuilder"/> to configure.</param>
+        /// <returns>The configured <see cref="IFunctionsWorkerApplicationBuilder"/>.</returns>
+        public static IFunctionsWorkerApplicationBuilder UseFunctionExecutionMiddleware(this IFunctionsWorkerApplicationBuilder builder)
```

Comment:
Might be confusing if both `UseFunctionExecutionMiddleware` and `UseDefaultWorkerMiddleware` are public, thoughts on just exposing the default? "FunctionExecutionMiddleware" sounds important enough that I may try to use both.

Reply:
I tried to make sure the documentation is pretty explicit about what gets registered with the default. The main reason I want to keep this is because it gives people the ability to compose with their own middleware. If they want to register theirs before the execution but after the output binding, the individual registration calls gives them that flexibility.

Reply:
is UseOutputBindingsMiddleware (and UseMiddleware) meant to be internal?

Reply:
@fabiocav, the concern I have here is that, if people update their existing project references to the new worker version, nothing would through, and their output bindings would just stop working. As this function now doesn't register that, which makes sense.

I am not sure what the alternative should be though. This may be Ok moving from preview, but wanted to bring it up, if you had any thoughts.

Reply:
These kinds of breaking changes need to be made prior to GA. Unfortunately, the previous middleware registration was overloaded, which shouldn't have been. One of the options is to cause a compilation break by renaming the method. We also need to update the templates and, as discussed, the registration should be further simplified with a simple call that adds all the defaults.

For now, since we do expect the output binding one to be registered, we can also cause an error if that isn't present. A few options here, but we shouldn't keep an API bug introduced in preview. We can have detailed information in the release notes.

Reply:
Discussed offline for the breaking change issue, and I agree with Fabio that it makes most sense to update this, and callout in release notes.

Sorry @mhoeger, I accidentally used your comment thread for my question on a separate topic. But to answer your question, I don't think it should be internal as it gives people ability to setup differently than how do we by default. For example, if they want to add a middleware between ExecutionMiddleware and OutputBindingMiddleware. They can call these in some different order.